fence/resources/audit/utils.py

```diff
@@ -88,11 +93,54 @@ def create_audit_log_for_request(response):
     return response
 
 
+def create_log_for_request(request: Request):
+    """
+    Right before processing the request (see `enable_local_audit_logging` decorator),
+    record an audit log. The data we need to record the logs are stored in
+    `flask.g.audit_data` before reaching this code.
+    """
+    username = get_current_user().username
+    method = request.method
+    endpoint = request.path
+    audit_data = getattr(flask.g, "audit_data", {})
+    request_url = endpoint
+    if request.query_string:
+        # could use `request.url` but we don't want the root URL
+        request_url += f"?{request.query_string.decode('utf-8')}"
+    request_url = _clean_authorization_request_url(request_url)
+    logger.info(
+        f"Request log: admin method called by user. Details: user=%s, method=%s, endpoint=%s, audit_data=%s, request_url=% ",
```

Comment:
```suggestion
        f"Incoming request: user=%s, method=%s, endpoint=%s, audit_data=%s, request_url=% ",
```
Since you made it generic, let's not assume it's an admin endpoint